transport/http/header.go

```diff
@@ -59,9 +59,16 @@ func (hm headerMapper) ToHTTPHeaders(from transport.Headers, to http.Header) htt
 func (hm headerMapper) FromHTTPHeaders(from http.Header, to transport.Headers) transport.Headers {
 	prefixLen := len(hm.Prefix)
 	for k := range from {
+		// Convert headers with Rpc-Header- prefix
 		if strings.HasPrefix(k, hm.Prefix) {
 			key := k[prefixLen:]
 			to = to.With(key, from.Get(k))
+			continue
+		}
+		// Preserve tracing headers
+		kLower := strings.ToLower(k)
+		if kLower == strings.ToLower(UberTraceIDHeader) || strings.HasPrefix(kLower, strings.ToLower(UberCtxHeader)) {
+			to = to.With(k, from.Get(k))
```

Comment:
With the existing approach, `uber-trace-id` and `uberctx-*` headers are not available inside application handlers. With the proposed approach, they become available, which may be a problem:
- Many handlers blindly copying headers (gateways, proxy service), i.e. they will unintentionally spread data to a following downstream services, which may be cause an incident.
- We'll open the door for ad-hoc parsing of tracing headers instead of tracing API usage

So, can we create tracing context on a higher level, when all request headers are available, and then remove those headers from request? That's how it was working before, right?